src/cascadia/WindowsTerminal/WindowEmperor.cpp

```diff
@@ -738,30 +737,14 @@ void WindowEmperor::_createMessageWindow(const wchar_t* className)
     StringCchCopy(_notificationIcon.szTip, ARRAYSIZE(_notificationIcon.szTip), appNameLoc.c_str());
 }
 
-// Counterpart to _postQuitMessageIfNeeded:
-// If it returns true, don't close that last window, if any.
-// This ensures we persist the last window.
-bool WindowEmperor::_shouldSkipClosingWindows() const
-{
-    const auto globalSettings = _app.Logic().Settings().GlobalSettings();
-    const size_t windowLimit = globalSettings.ShouldUsePersistedLayout() ? 1 : 0;
-    return _windows.size() <= windowLimit;
-}
-
 // Posts a WM_QUIT as soon as we have no reason to exist anymore.
-// That basically means no windows [^1] and no message boxes.
-//
-// [^1] Unless:
-// * We've been asked to persist the last remaining window
-//   in which case we exit with 1 remaining window.
-// * We're allowed to be headless
-//   in which case we never exit.
+// That basically means no windows and no message boxes.
 void WindowEmperor::_postQuitMessageIfNeeded() const
 {
-    const auto globalSettings = _app.Logic().Settings().GlobalSettings();
-    const size_t windowLimit = globalSettings.ShouldUsePersistedLayout() ? 1 : 0;
-
-    if (_messageBoxCount <= 0 && _windows.size() <= windowLimit && !globalSettings.AllowHeadless())
+    if (
+        _messageBoxCount <= 0 &&
+        _windowCount <= 0 &&
+        !_app.Logic().Settings().GlobalSettings().AllowHeadless())

```

Comment:
between the lines... does this mean that `wt --help` when headless mode is enabled will leave one running?